### DISCUSSION TITLE
Where can I change the NMS parameters?

### DISCUSSION THREAD

**Discussion by mochiao** [ultralytics/ultralytics#1426] [Q&A] [answered]
As is mentioned in the title, I want to change the parameter of NMS. I've met some problems when I want to predict on the yolov8n model. Sometimes the Prediction produces 2-3 detections on the same object. How can I optimize it?

**Comment by Pranay-Pandey** [comment, marked as answer]
Change it while performing the prediction using the YOLO model object

Example - 
model = YOLO("yolov8n.pt")
results = model(file, conf=value, **iou=iou_threshold**)

**Reply by mochiao** [reply]
perfect! I appreciated. And maybe this parameter can be use while training?

**Reply by glenn-jocher** [reply]
@mochiao yes, you can adjust the NMS parameters during training as well. These parameters can be fine-tuned in the configuration file that you use for training your model. Look for the NMS settings within the configuration file, where you can set the `iou_threshold` and `conf_threshold` to optimize the performance of your model during training. Adjusting these values can help reduce multiple detections of the same object. Remember to validate your model after training to ensure that the changes have improved the model's detection capabilities. For more detailed guidance on training and configuration, you can refer to the training documentation on our Ultralytics Docs. 📘🚀
